internal/gitea/repo.go

```diff
@@ -0,0 +1,190 @@
+package gitea
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"code.gitea.io/sdk/gitea"
+	"github.com/gomarkdown/markdown"
+	"github.com/jnsgruk/releasegen/internal/repos"
+)
+
+const giteaReleasesPerRepo = 3
+
+// Repository represents a single repository. Note that this might be a Gitea
+// Repository or it might be a folder in a Gitea Repository if the Repository
+// is a monorepo.
+type Repository struct {
+	Details       repos.RepoDetails
+	org           string // The Gitea Org that owns the repo.
+	client        *gitea.Client
+	defaultBranch string
+	folder        string
+}
+
+// Process populates the Repository with details of its releases, and commits.
+func (r *Repository) Process() error {
+	// Iterate over the releases in the Gitea repo and add them to our repository's details.
+	err := r.processReleases()
+	if err != nil {
+		return err
+	}
+
+	if len(r.Details.Releases) > 0 {
+		// Calculate the number of commits since the latest release.
+		err := r.processCommitsSinceRelease()
+		if err != nil {
+			return err
+		}
+	} else {
+		// If there are no releases, get the latest commit instead.
+		err := r.processCommits()
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
It may not be any clearer, so up to you, but you could reuse `err` and put the error checking once outside the if/else:

```suggestion
	if len(r.Details.Releases) > 0 {
		// Calculate the number of commits since the latest release.
		err = r.processCommitsSinceRelease()
	} else {
		// If there are no releases, get the latest commit instead.
		err = r.processCommits()
	}
	if err != nil {
		return err
	}
```